source/spark_sql_migrations/tests/current_state/test_create_current_state.py

```diff
@@ -0,0 +1,89 @@
+﻿import pytest
+import spark_sql_migrations.current_state.create_current_state as sut
+
+from pyspark.sql import SparkSession
+from unittest.mock import Mock
+from tests.helpers.test_schemas import schema_config
+from tests.helpers.spark_helper import reset_spark_catalog
+
+
+def test_get_schema_scripts_matches_schema_config() -> None:
+    # Arrange
+    schemas = []
+    for schema in schema_config:
+        schemas.append(schema.name)
+
+    # Act
+    actual = sut._get_schema_scripts()
+
+    # Assert
+    assert len(actual) == len(schemas)
+
+
+def test_get_table_scripts_matches_schema_config() -> None:
+    # Arrange
+    tables = []
+    for schema in schema_config:
+        for table in schema.tables:
+            tables.append(table.name)
+
+    # Act
+    actual = sut._get_table_scripts()
+
+    # Assert
+    assert len(actual) == len(tables)
+
+
+def test_create_all_tables_creates_all_tables(spark: SparkSession) -> None:
+    # Arrange
+    reset_spark_catalog(spark)
+
+    # Act
+    sut.create_all_tables()
+
+    # Assert
+    for schema in schema_config:
+        for table in schema.tables:
+            assert spark.catalog.tableExists(table.name, schema.name)
+
+
+def test_create_all_tables_when_table_is_missing_it_should_create_the_missing_tables(
```

Comment:
In wholesale we use double underscore like this:
test__create_all_tables__when_table_is_missing__creates_the_missing_tables

I think it makes it easier to read. I'm not sure what style we used in shared repos, though.

Reply:
hmm, that makes sense. I will introduce this for the Migrations repo as well.